client/src/main/java/org/eclipse/hono/client/HonoClient.java

```diff
@@ -281,4 +286,109 @@
      *         AMQP <em>open</em> frame, {@code false} otherwise.
      */
     boolean supportsCapability(Symbol capability);
+
+    /**
+     * Builder for for HonoClient instances.
+     * <p>
+     * This class is not thread safe.
+     */
+    class HonoClientBuilder {
```

Comment:
what value does this builder add? it accepts the same properties as the standard constructor ...

Reply:
Yes, but builder can be extended without breaking the API. Constructors cannot and thus you end up with a bunch of constructors over time. But yes, we could collapse the two builders into one.

Reply:
FMPOV the purpose of the *builder* is already covered by the config properties. This, BTW, seems to also be the way pursued by vert.x in all of its APIs ...

Reply:
Ok, do you want me to drop both builders then?

Reply:
I'd appreciate that